# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/naga/src/front/spv/convert.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/naga/src/front/spv/convert.rs b/third_party/rust/naga/src/front/spv/convert.rs
--- a/third_party/rust/naga/src/front/spv/convert.rs
+++ b/third_party/rust/naga/src/front/spv/convert.rs
@@ -127,37 +127,42 @@ pub(super) fn map_builtin(word: spirv::W
         Some(Bi::ClipDistance) => crate::BuiltIn::ClipDistance,
         Some(Bi::CullDistance) => crate::BuiltIn::CullDistance,
         Some(Bi::InstanceIndex) => crate::BuiltIn::InstanceIndex,
         Some(Bi::PointSize) => crate::BuiltIn::PointSize,
         Some(Bi::VertexIndex) => crate::BuiltIn::VertexIndex,
         // fragment
         Some(Bi::FragDepth) => crate::BuiltIn::FragDepth,
         Some(Bi::FrontFacing) => crate::BuiltIn::FrontFacing,
+        Some(Bi::PrimitiveId) => crate::BuiltIn::PrimitiveIndex,
         Some(Bi::SampleId) => crate::BuiltIn::SampleIndex,
         Some(Bi::SampleMask) => crate::BuiltIn::SampleMask,
         // compute
         Some(Bi::GlobalInvocationId) => crate::BuiltIn::GlobalInvocationId,
         Some(Bi::LocalInvocationId) => crate::BuiltIn::LocalInvocationId,
         Some(Bi::LocalInvocationIndex) => crate::BuiltIn::LocalInvocationIndex,
         Some(Bi::WorkgroupId) => crate::BuiltIn::WorkGroupId,
         Some(Bi::WorkgroupSize) => crate::BuiltIn::WorkGroupSize,
+        Some(Bi::NumWorkgroups) => crate::BuiltIn::NumWorkGroups,
         _ => return Err(Error::UnsupportedBuiltIn(word)),
     })
 }
 
 pub(super) fn map_storage_class(word: spirv::Word) -> Result<super::ExtendedClass, Error> {
     use super::ExtendedClass as Ec;
     use spirv::StorageClass as Sc;
     Ok(match Sc::from_u32(word) {
         Some(Sc::Function) => Ec::Global(crate::StorageClass::Function),
         Some(Sc::Input) => Ec::Input,
         Some(Sc::Output) => Ec::Output,
         Some(Sc::Private) => Ec::Global(crate::StorageClass::Private),
         Some(Sc::UniformConstant) => Ec::Global(crate::StorageClass::Handle),
-        Some(Sc::StorageBuffer) => Ec::Global(crate::StorageClass::Storage),
+        Some(Sc::StorageBuffer) => Ec::Global(crate::StorageClass::Storage {
+            //Note: this is restricted by decorations later
+            access: crate::StorageAccess::all(),
+        }),
         // we expect the `Storage` case to be filtered out before calling this function.
         Some(Sc::Uniform) => Ec::Global(crate::StorageClass::Uniform),
         Some(Sc::Workgroup) => Ec::Global(crate::StorageClass::WorkGroup),
         Some(Sc::PushConstant) => Ec::Global(crate::StorageClass::PushConstant),
         _ => return Err(Error::UnsupportedStorageClass(word)),
     })
 }